external/build/youtube-playlist.js

```diff
@@ -0,0 +1,204 @@
+/**
+ * @fileoverview Fetches the most recent playlists by specified channels from YouTube
+ */
+
+require('dotenv').config();
+
+const fs = require('fs').promises;
+const path = require('path');
+const ms = require('ms');
+const {google} = require('googleapis');
+
+/**
+ * Time interval to determine if the current data set is stale
+ */
+const FETCH_INTERVAL = ms('4h');
+
+/**
+ * Config constants to interact with the YouTube API
+ */
+const API_KEY = process.env.YOUTUBE_API_KEY;
+const PART = ['snippet'];
+const MAX_RESULTS = 50;
+
+/**
+ * Set Google Chrome Developers Youtube channel ID (https://www.youtube.com/@ChromeDevs)
+ */
+const CHANNELS = ['UCnUYZLuoy1rq1aVMwx4aTzw'];
+
+/**
+ * Target directory for the fetched data
+ */
+const targetFile = path.join(__dirname, '../data/youtube-playlist.json');
+const currentTimestamp = Date.now();
+
+const result = {};
+
+const youtube = google.youtube('v3');
+
+/**
+ * Starts the data fetching process form the channel name
+ * specified in the config.
+ */
+
+async function run() {
+  if (!process.env.YOUTUBE_API_KEY) {
+    throw new Error('No `YOUTUBE_API_KEY` environment var for production');
+  }
+
+  /**
+   * Fetch current data set and checks timestamp to see if the data is stale
+   */
+  let dataAge = 0;
+  try {
+    const currentDataRaw = await fs.readFile(targetFile, {encoding: 'utf8'});
+    const currentData = JSON.parse(currentDataRaw);
+
+    dataAge = currentTimestamp - currentData.timestamp;
+  } catch (error) {
+    console.warn(
+      'No existing data found or existing data malformatted. Fetching from YouTube...'
+    );
+  }
+  if (dataAge > FETCH_INTERVAL || !dataAge) {
```

Comment:
The following block is fairly small but it's usually good practice to inverse the condition and early exit, so

```js
if (dataAge && dataAge < FETCH_INTERVAL) {
  console.log(`YouTube data is ${ms(dataAge)} old, no need to fetch.`);
  return;
}
```